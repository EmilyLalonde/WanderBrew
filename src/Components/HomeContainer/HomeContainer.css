h3 {
  font-size: 35px;
  margin: 0;
  margin-bottom: 1%;
  margin-top: 4%;
  text-align: center;
}