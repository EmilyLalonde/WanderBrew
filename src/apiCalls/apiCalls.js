export const getBreweriesByState = async (state) => {
  const stateBreweries = `https://api.openbrewerydb.org/breweries?by_state=${state}&per_page=50`
  const response = await fetch(stateBreweries)
  if(!response.ok) {
    throw new Error('There was an error getting your albums')
  }
  const allStateBreweries = await response.json();
  return allStateBreweries
}

// export const getBreweriesByCity = async (city) => {
//   const cityBreweries = `https://api.openbrewerydb.org/breweries?by_name=${city}&per_page=50`
//   const response = await fetch(cityBreweries)
//   if(!response.ok) {
//     throw new Error('There was an error getting your albums')
//   }
//   const allCityBreweries = await response.json();
//   return allCityBreweries
// }


