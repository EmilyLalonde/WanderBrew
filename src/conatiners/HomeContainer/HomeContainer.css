h3 {
  font-size: 35px;
  margin: 0;
  margin-bottom: 1%;
  margin-top: 2%;
  text-align: center;
}