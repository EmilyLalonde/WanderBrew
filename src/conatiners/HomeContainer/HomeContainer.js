import React from 'react'
import { connect } from 'react-redux';
import './HomeContainer.css'
import DefaultBreweriesContainer from '../../Components/DefaultBreweriesContainer/DefaultBreweriesContainer'

export const HomeContainer = (props) => {
    return (
    <div>
      <h3>Featured Denver Breweries</h3>
      <DefaultBreweriesContainer breweries={props.denverResults}/>
      <h3>Featured National Breweries</h3>
      <DefaultBreweriesContainer breweries={props.nationalResults}/>
    </div>
    )
}

export const mapStateToProps = state => ({
  denverResults: state.denverResults,
  nationalResults: state.nationalResults
})

export default connect(mapStateToProps, null)(HomeContainer)