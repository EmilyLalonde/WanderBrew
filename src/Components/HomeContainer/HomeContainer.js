import React, { Component } from 'react'
import { getPopularDenverBreweries, getPopularBreweriesNational } from '../../apiCalls/apiCalls'
import './HomeContainer.css'
import DefaultBreweriesContainer from '../DefaultBreweriesContainer/DefaultBreweriesContainer'

class HomeContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      denverBreweries: [],
      nationalBreweries: []
    }
  }

  componentDidMount () {
      getPopularDenverBreweries()
      .then(data => this.setState({denverBreweries: data}))
      getPopularBreweriesNational()
      .then(data => this.setState({nationalBreweries: data}))
  }

  render() {
    return (
    <div>
      <DefaultBreweriesContainer breweries={this.state.denverBreweries}/>
      <DefaultBreweriesContainer breweries={this.state.nationalBreweries}/>
    </div>
    )
  }
}

export default HomeContainer