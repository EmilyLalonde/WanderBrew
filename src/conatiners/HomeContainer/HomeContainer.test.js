import React from 'react';
import { shallow } from 'enzyme'
import { HomeContainer, mapStateToProps } from './HomeContainer';
import { getBreweriesByState, getBreweriesByName } from "../../apiCalls/apiCalls";

jest.mock('../../apiCalls/apiCalls');

getBreweriesByState.mockImplementation(() => {
  return Promise.resolve({
    id: 1516, 
    name: 'Ratio Beerworks', 
    street: '2920 Larimer St', 
    city: 'Denver', 
    state:'Colorado', 
    phone:'3012669264'
  })
});

getBreweriesByName.mockImplementation(() => {
  return Promise.resolve({
    id: 1516, 
    name: 'Ratio Beerworks', 
    street: '2920 Larimer St', 
    city: 'Denver', 
    state:'Colorado', 
    phone:'3012669264'
  })
});

describe('HomeContainer', () => {
  let wrapper;

  let mockDenverResults = [
    {
      id: 1516, 
      name: 'Ratio Beerworks', 
      street: '2920 Larimer St', 
      city: 'Denver', 
      state:'Colorado', 
      phone:'3012669264'
    }
  ]

  let mockNationalResults = [
    {
      id: 1516, 
      name: 'Ratio Beerworks', 
      street: '2920 Larimer St', 
      city: 'Denver', 
      state:'Colorado', 
      phone:'3012669264'
    }
  ]

  let mockState = {
    denverResults: [mockDenverResults],
    nationalResults: [mockNationalResults]
  }

  beforeEach(() => {
    wrapper = shallow (<HomeContainer props={mockDenverResults || mockNationalResults}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

it('should call mapStateToProps with the props it needs', () => {
  let mockDenverResults = [
    {
      id: 1516, 
      name: 'Ratio Beerworks', 
      street: '2920 Larimer St', 
      city: 'Denver', 
      state:'Colorado', 
      phone:'3012669264'
    }
  ]

  let mockNationalResults = [
    {
      id: 1516, 
      name: 'Ratio Beerworks', 
      street: '2920 Larimer St', 
      city: 'Denver', 
      state:'Colorado', 
      phone:'3012669264'
    }
  ]

  let mockState = {
    denverResults: [mockDenverResults],
    nationalResults: [mockNationalResults]
  }

  const expected = {
    denverResults: [mockDenverResults],
    nationalResults: [mockNationalResults],
  }
  const mappedProps = mapStateToProps(mockState)
  
  expect(mappedProps).toEqual(expected);
})