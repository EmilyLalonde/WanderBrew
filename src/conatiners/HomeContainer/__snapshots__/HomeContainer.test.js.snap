// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer should match the snapshot 1`] = `
<div>
  <h3>
    Featured Denver Breweries
  </h3>
  <DefaultBreweriesContainer />
  <h3>
    Featured National Breweries
  </h3>
  <DefaultBreweriesContainer />
</div>
`;
